app/src/main/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapper.kt

```diff
@@ -0,0 +1,50 @@
+package com.hedvig.app.feature.profile.ui.tab
+
+import com.hedvig.android.owldroid.graphql.ProfileQuery
+import com.hedvig.app.common.Mapper
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.util.LocaleManager
+import com.hedvig.app.util.apollo.format
+import com.hedvig.app.util.apollo.toMonetaryAmount
+import com.hedvig.app.util.featureflags.FeatureManager
+import com.hedvig.app.util.featureflags.flags.Feature
+import java.util.Locale
+
+class ProfileQueryDataToProfileUiStateMapper(
+    private val featureManager: FeatureManager,
+    private val marketManager: MarketManager,
+    private val localeManager: LocaleManager,
+) : Mapper<ProfileQuery.Data, ProfileUiState> {
+
+    override suspend fun map(from: ProfileQuery.Data): ProfileUiState {
+        val cashbackFragment = from.cashback?.fragments?.cashbackFragment
+        val charityState = run {
+            if (featureManager.isFeatureEnabled(Feature.SHOW_CHARITY).not()) return@run CharityState.DontShow
+            val charityName = cashbackFragment?.name
+            if (charityName != null) return@run CharityState.Selected(charityName)
+            CharityState.NoneSelected
+        }
```

Comment:
Is this use of `run` a bit weird? This was the only simple way I found of wanting to do a simple val assignment but with some logic in there. I am thinking it may be weird because I've never had to use this before, but maybe there simply wasn't a case where I needed it 🤷‍♂️

Reply:
I think it's sensible.